3218 2 change operation type

BCerki opened the pull request:
card #3218 

This PR:
- Adding a confirmation modal for operation type changes in the registration form.
- Modifying tests to validate the new confirmation behavior.
- Updating the operation service and facility timeline service to support deletion of facilities upon operation type changes.
- In admin operation type is readonly, and there's a backend check to make sure you can't change the type if the operation's status is registered
- Sonar fail is not a real security issue. We're only using the random number to generate a react key, same strategy as for the registration purpose modal

andrea-williams commented:
I accidentally broke this in my local. I'm not even entirely sure what I did to break it, but now I'm stuck.

I thiiiink (?) these were my steps to reproduce:
1. selected Brine LFO and started registration
2. Brine only has 1 mock facility, I created multiple
3. Navigated back to first step of registration and changed operation type to SFO
4. Proceeded to next step of form (Facilities page), where the facilities grid was displayed (I guess technically it shouldn't have, since it's now an SFO, but I wouldn't consider that something worth worrying about at this time). There were 0 records in the grid
5. Clicked the button to add a new facility and hit Save
6. I think then I went back to Step 1 of the form, didn't change any data, then clicked Next to go back to Step 2
7. At this point the facility I created in 5 wasn't shown in the grid, instead I saw the page I was expecting (to create a facility for the SFO), but then when I tried to save the page I got this error. Basically the first facility I created is in the DB but not appearing on the front end. Now I'm stuck and can't proceed with registration.

![Screenshot 2025-05-12 at 3 23 24 PM](https://github.com/user-attachments/assets/281dce54-f738-4523-8084-b2db524c51b0)

BCerki commented:
> I accidentally broke this in my local. I'm not even entirely sure what I did to break it, but now I'm stuck.
> 
> I thiiiink (?) these were my steps to reproduce:
> 
> 1. selected Brine LFO and started registration
> 2. Brine only has 1 mock facility, I created multiple
> 3. Navigated back to first step of registration and changed operation type to SFO
> 4. Proceeded to next step of form (Facilities page), where the facilities grid was displayed (I guess technically it shouldn't have, since it's now an SFO, but I wouldn't consider that something worth worrying about at this time). There were 0 records in the grid
> 5. Clicked the button to add a new facility and hit Save
> 6. I think then I went back to Step 1 of the form, didn't change any data, then clicked Next to go back to Step 2
> 7. At this point the facility I created in 5 wasn't shown in the grid, instead I saw the page I was expecting (to create a facility for the SFO), but then when I tried to save the page I got this error. Basically the first facility I created is in the DB but not appearing on the front end. Now I'm stuck and can't proceed with registration.
> 
> ![Screenshot 2025-05-12 at 3 23 24 PM](https://private-user-images.githubusercontent.com/54914403/442966538-281dce54-f738-4523-8084-b2db524c51b0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcwOTAwMzAsIm5iZiI6MTc0NzA4OTczMCwicGF0aCI6Ii81NDkxNDQwMy80NDI5NjY1MzgtMjgxZGNlNTQtZjczOC00NTIzLTgwODQtYjJkYjUyNGM1MWIwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTEyVDIyNDIxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg0MmM1ZWZjYzM5YTBjMDg2YTAyZDc0NDZhOWEzODg3YjU0YWU0NzdjYWY3MGM0ZGYyMGY1Nzk3NGUyOGFmZGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.KHs7g52czeU5lD690hvV9mrpjk5lVaRbUI49v72B7bM)

Ahhh, I think this is because I introduced a bug in a previous PR to do with facilities: https://github.com/bcgov/cas-registration/pull/3342/ Once that's in, I'll rebase this one, and hopefully that will fix it

andrea-williams reviewed (commented):
Have tested and confirmed the behaviour of the PR matches the ticket description.

I did notice that my document upload isn't working - I uploaded the required docs as part of the registration workflow, and am able to register succesfully, but when I go back to Operation Details page, it says that no docs were uploaded. Not sure if that's related to this PR or if I just mucked something up in my local settings. 

Would you mind checking in your local? I won't worry about it if it's just a me-problem.

BCerki commented:
> Have tested and confirmed the behaviour of the PR matches the ticket description.
> 
> I did notice that my document upload isn't working - I uploaded the required docs as part of the registration workflow, and am able to register succesfully, but when I go back to Operation Details page, it says that no docs were uploaded. Not sure if that's related to this PR or if I just mucked something up in my local settings.
> 
> Would you mind checking in your local? I won't worry about it if it's just a me-problem.

It's not just you; it's a problem for everyone in local: https://github.com/bcgov/cas-registration/issues/3252. You can change your `env` if you want the attachments to stick around
